corehq/apps/reports/views.py

```diff
@@ -2091,14 +2155,18 @@ def export_report(request, domain, export_hash, format):
         meta = db.metadb.get(parent_id=export_hash, key=export_hash)
     except models.BlobMeta.DoesNotExist:
         return report_not_found
+
+    if domain != meta.domain:
+        raise Http404()
+
     report_class = meta.properties["report_class"]
 
     try:
         report_file = db.get(export_hash, type_code=CODES.tempfile)
     except NotFound:
         return report_not_found
     with report_file:
-        if not request.couch_user.has_permission(domain, 'view_report', data=report_class):
+        if not request.couch_user.has_permission(meta.domain, 'view_report', data=report_class):
```

Comment:
The code as it was before seems more correct. I think it doesn't matter, given check and raise 404 above, but might be worth reverting.
```suggestion
        if not request.couch_user.has_permission(domain, 'view_report', data=report_class):
```

Reply:
Ah, you're right -- after ensuring that the domain and meta domain match, there's really no reason to use the meta domain anymore